src/chess/bitboard.cc

```diff
@@ -303,17 +303,17 @@ Move::Move(const std::string& str, bool black) {
 }
 
 uint16_t Move::as_packed_int() const {
-  if (promotion_ == Promotion::Knight) {
-    return from_.as_int() * 64 + to_.as_int();
+  if (promotion() == Promotion::Knight) {
+    return from().as_int() * 64 + to().as_int();
   } else {
-    return static_cast<int>(promotion_) * 64 * 64 + from_.as_int() * 64 +
-           to_.as_int();
+    return static_cast<int>(promotion()) * 64 * 64 + from().as_int() * 64 +
+           to().as_int();
   }
 }
```

Comment:
Since the knight promotion is alone on bit 14 (0bxNxxxxxxxxxxxxxx, and none, queen, rook, and bishop are on bits 12 and 13) and the castling is on bit 15, I suggest:
```
uint16_t Move::as_packed_int() const {
  return data_ & 0b0011111111111111;
}
```

Reply:
There's another change which fully alters the as_packed_int and as_nn_index to just return data_, and also modifies BuildMoveIndecies accordingly, but I'd rather keep this change as a single PR and do the stuff modifiying those functions in a separate PR

Reply:
Sounds good.
Will those changes require changes to the NN policy head output as well?

Reply:
No, the only changes needed are to bitboard.cc. It changes Build_move_indices so that kMoveToIdx[data_] returns the same thing as the current function, so that the NN sees the same inputs and returns the same outputs